### PR TITLE
Chore/improved functions

### PR DESCRIPTION
**Pull request by Yagth**
## Description

I have added comments for each transformation function about what it is doing. The documentation is mostly taken from the holman paper and modified to match the behavior of the functions. Also, other minor updates like formatting, function naming and some function locations have been made. The functions that were defined in the `rte-helpers.metta` file but are only used in a single file have been moved to that file.

I have also added two more test cases that were edge cases for the `promote-common-constraint` and `zero-constraint-subsumption` transformation rules. They are commented out for now but a fix is under way.

I have also add few helper functions that will be useful for rte implementation.

## Motivation and Context

The main motivation of this context is to keep the code consistency and improved code readability.

## How Has This Been Tested?

I run all the test cases locally and everything is working fine after the update.

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: utilities/general-helpers.metta [hunk 1]
```diff
@@ -57,3 +57,7 @@
 (= (findAndReplace $old $new $list)
     (collapse (replace $old $new (superpose $list)))
 )
+
+
+(: until (-> (-> $a Bool) (-> $a $a) $a $a))
+(= (until $predicate $function $x) (if ($predicate $x) $x (let $x' ($function $x) (until $predicate $function $x'))))
```
Comment (Nahom32) [inline utilities/general-helpers.metta:63]: @Yagth is this implementation concerned with executing a function until a truth value is met?

File: utilities/tests/general-helper-functions-test.metta [hunk 2]
```diff
@@ -38,3 +38,59 @@
...
+;; These are functions used for testing until function.
+
+;; It returns True only if the input is zero.
+(: isZero (-> Number Bool))
+(= (isZero $x) (== $x 0))
+
+;; It will return True if greater than 10.
+(: greaterThanTen (-> Number Bool))
+(= (greaterThanTen $x) (> $x 10))
+
```
Comment (Nahom32) [inline utilities/tests/general-helper-functions-test.metta:53]: why do you need this function? I mean it is a trivial check and very specific

**Review by Nahom32 — CHANGES_REQUESTED** [review]

File: utilities/general-helpers.metta (hunk 1 above)
Comment (Yagth) [reply utilities/general-helpers.metta:63]: Yep. I was planning to use it like a while loop for some cases if possible

File: utilities/tests/general-helper-functions-test.metta (hunk 2 above)
Comment (Yagth) [reply utilities/tests/general-helper-functions-test.metta:53]: Yeah it is. I only need it to for testing the `until` function, hence the reason I put it in the test file.

**Review by Nahom32 — APPROVED** [review]
